larreco/Calorimetry/Calorimetry_module.cc

```diff
@@ -76,6 +102,7 @@ namespace calo {
     bool fUseArea;
     bool fSCE;
     bool fFlipTrack_dQdx; //flip track direction if significant rise of dQ/dx at the track start
+    double fNotOnTrackZcut; ///< Exclude trajectory points with _z_ lower than this [cm]
```

Comment:
Suggest `double const` as this variable need not be modified in the lifetime of the object.

Reply:
Hold on... that does not look like the latest commit in the branch.
Has something been lost in the process?

Reply:
I just realised that `std::optional` can host a constant, and that constant can be constructed at any time (as opposed to only when the `std::optional` object is constructed).
So your suggestion _is_ possible with the new code:
```suggestion
    std::optional<double const> fNotOnTrackZcut; ///< Exclude trajectory points with _z_ lower than this [cm]
```
A world of constantness abuse has just opened in front of my eyes.

Reply:
@PetrilloAtWork What's the benefit in the data member not being const?

Reply:
None.
But I do not see how to initialise it without an helper function, since it requires a different constructor depending on the parameter value (please check the latest version).
Something like:
```
namespace details {
  template <typename T>
  std::optional<T> get_optional
    (fhicl::ParameterSet const& pset, std::string const& key)
  {
    std::optional<T> param;
    if (pset.has_key(key)) param.emplace(pset.get<T>(key));
    return param;
  } // get_optional()
} // namespace details
```
A bit of an overkill, in my opinion.
You can also consider adding a `fhicl::ParameterSet::get_optional()`, I guess.
Or maybe I am missing some relevant existing feature.

Reply:
I'd probably favor something like:

````c++
Calorimetry(ParameterSet const& pset)
  : ...
  , fNotOnTrackZcut{
      pset.has_key(...) ? std::make_optional(pset.get<double>(...)) : std::nullopt}
{}
````

Reply:
I like it.